packages/core/src/components/graphs/treemap.ts

```diff
@@ -199,7 +200,7 @@ export class Treemap extends Component {
 					return [
 						{
 							text: d.data.name,
-							color: color.l < 0.5 ? 'white' : 'black'
+							backgroundColor: color
```

Comment:
seems like you removed the `color` field here. is there a reason for that?

Reply:
The only reason I removed it was because it didn't seem to be utilized anywhere; if this is not the case I can revert this change.

Reply:
that sets the text fill based on backgroundColor

Reply:
I think the `color` property within the text node's `data` wasn't being used; instead the text node's color is set by the `textFillColor` function which utilizes either the new `backgroundColor` property that I've added or the fill color of the parent `rect.leaf` node.

The main issue was that when the `textFillColor` called `getComputedStyles` on the `rect.leaf` node, the `fill` property isn't set to the custom color early enough, so it always ended up determining the text color based on the default colors.

Reply:
yes @SMassola the text color values seem to be getting set properly in cases where you're providing custom colors, but if you don't, it seems like your PR has broken that case

Reply:
> yes @SMassola the text color values seem to be getting set properly in cases where you're providing custom colors, but if you don't, it seems like your PR has broken that case

You're right, I've modified the code to send the `backgroundColor` in the proper format (simply a string) to the `textFillColor` function.